api/src/pcapi/core/users/api.py

```diff
@@ -940,12 +940,41 @@ def update_notification_subscription(
     if subscriptions is None:
         return
 
+    old_subscriptions = user.get_notification_subscriptions()
     user.notificationSubscriptions = {
         "marketing_push": subscriptions.marketing_push,
         "marketing_email": subscriptions.marketing_email,
         "subscribed_themes": subscriptions.subscribed_themes,
     }
 
+    logger.info(
+        "Notification subscription update",
+        extra={
+            "analyticsSource": "app-native",
+            "newlySubscribedTo": {
+                "email": subscriptions.marketing_email and not old_subscriptions.marketing_email,
+                "push": subscriptions.marketing_push and not old_subscriptions.marketing_push,
+                "themes": [
+                    theme
+                    for theme in subscriptions.subscribed_themes
+                    if theme not in old_subscriptions.subscribed_themes
+                ],
```

Comment:
Remarque : ça peut s'écrire assez simplement avec les `set` : 
```suggestion
                "themes": list(set(subscriptions.subscribed_themes) - set(old_subscriptions.subscribed_themes)),
```

Reply:
On peut même omettre la conversion avec `list` et passer le set à la fonction de log. Elle [saura](https://github.com/pass-culture/pass-culture-main/blob/dff41a91f3882e5ec33cfc697e618e66e86fcf58/api/src/pcapi/core/logging.py#L144-L145) quoi en faire.